piecrust/src/vm.rs

```diff
@@ -125,9 +133,9 @@ impl VM {
     ///
     /// [`Session`]: Session
     /// [`session`]: VM::session
-    pub fn genesis_session(&self) -> Session {
+    pub fn genesis_session(&self, data: SessionData) -> Session {
```

Comment:
Could the commit `base` also be in `SessionData`? In this way, we could even get rid of the `genesis_session()` in favor of just `session()`. It also starts feeling like the same builder pattern as `ModuleData` should be applied here.